app.js

```diff
@@ -0,0 +1,161 @@
+const express = require('express');
+const app = express();
+var methodOverride = require('method-override');
+const bp = require('body-parser');
+const exphbs = require('express-handlebars');
+const Products = require('./db/products.js');
+const PE_Inventory = new Products();
+const Articles = require('./db/articles.js');
+const Art_Inventory = new Articles();
+
+
+app.use(express.static('public'));
+
+app.use(bp.urlencoded({ extended: true }));
+
+app.use(methodOverride('_method'));
+
+app.engine('.hbs', exphbs({ defaultLayout: 'main', extname: '.hbs' }));
+app.set('view engine', '.hbs'); 
+
+// render all product/article items
+app.get('/', (req, res) => {
+  // const items = PE_Inventory.all();
+  // const artItem = Art_Inventory.all();
+  res.render('home');
+});
+
+
+// render out the product form
+app.get('/product/new', (req, res) => {
+  res.render('productform');
+});
+
+//render out the article form
+app.get('/article/new', (req, res) => {
+    res.render('articleform');
+});
+
+
+//render out the products home page
+app.get('/productshome', (req, res) => {
+  const detailItems = PE_Inventory.all();
+  console.log('product details CL', { detailItems }) //don't understand how this is displayed in console and what it does below
+  res.render('productshome', { detailItems });
+})
+
+//render out the articles home page
+app.get('/articleshome', (req, res) => {
+  const articleItems = Art_Inventory.all();
+  res.render('articleshome', { articleItems });
+})
+
+// render out product details
+app.get('/product/:id', (req, res) => {
+  // console.log('did it call');
+  const { id } = req.params;
+  const item = PE_Inventory.getItemById(id);
+  // console.log('item', item);
+  res.render('productdetail', item);
+});
+
+
+//render out article details
+app.get('/article/:title', (req, res) => {
+  const { title } = req.params;
+  const item = Art_Inventory.getItemById(title);
+  console.log('item CS', item)
+  res.render('articledetails', item);
+});
+
+//render out products edit get
+app.get('/products/:id/edit', (req, res) => {
+  const { id } = req.params;
+  let productToEdit = PE_Inventory.getItemById(id);
+  res.render('edit', { productToEdit });
+});
+
+//render out articles edit get
+app.get('/articles/:id/edit', (req, res) => {
+  console.log('am I getting called')
+  const { id } = req.params;
+  let articleToEdit = Art_Inventory.getItemById(id);
```

Comment:
`let`, is a signal that the variable may be reassigned, such as a counter in a loop, or a value swap in an algorithm. It also signals that the variable will be used only in the block it’s defined in, which is not always the entire containing function. - https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75